builtin/providers/google/resource_compute_router_peer_test.go

```diff
@@ -0,0 +1,263 @@
+package google
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+
+	"google.golang.org/api/compute/v1"
+)
+
+func TestAccComputeRouterPeer_basic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccComputeRouterPeer_basic,
+				Check: testAccCheckComputeRouterPeerExists(
+					"google_compute_router_peer.foobar"),
+			},
+			resource.TestStep{
+				Config: testAccComputeRouterPeer_keepRouter,
+				Check: testAccCheckComputeRouterPeerDestroy(
+					"google_compute_router_peer.foobar"),
+			},
+		},
+	})
+}
+
+func testAccCheckComputeRouterPeerDestroy(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		config := testAccProvider.Meta().(*Config)
+		project := config.Project
+
+		routersService := compute.NewRoutersService(config.clientCompute)
+
+		for _, rs := range s.RootModule().Resources {
+			if rs.Type != "google_compute_router_peer" {
+				continue
+			}
+
+			region := rs.Primary.Attributes["region"]
+			name := rs.Primary.Attributes["name"]
+			routerName := rs.Primary.Attributes["router"]
+
+			router, err := routersService.Get(project, region, routerName).Do()
+
+			if err != nil {
+				return fmt.Errorf("Error Reading Router %s: %s", routerName, err)
+			}
+
+			var peerExists bool = false
+
+			var peers []*compute.RouterBgpPeer = router.BgpPeers
+			for _, peer := range peers {
+
+				if peer.Name == name {
+					peerExists = true
+					break
+				}
+			}
+
+			if peerExists {
+				return fmt.Errorf("Peer %s still exists on router %s", name, router.Name)
+			}
+
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckComputeRouterPeerExists(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No ID is set")
+		}
+
+		config := testAccProvider.Meta().(*Config)
+		name := rs.Primary.Attributes["name"]
+		routerName := rs.Primary.Attributes["router"]
+		region := rs.Primary.Attributes["region"]
```

Comment:
Would really like to see this honor the same logic as `getRegion`, just like the resource.